packages/template/vite/tmpl/forge.config.js

```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  packagerConfig: {},
```

Comment:
If #3077 gets merged, this will need to be updated as well to enable Asar by default.

Reply:
Okay!

Reply:
> In my opinion, the WebpackPluginConfig design could use some improvement. For example, why are config.mainConfig and config.renderer.config named that way? It's not very intuitive.

I'm worried about users using Webpack templates, who are used to the Webapck template configuration.
If I were to redesign the configuration of the Vite template, it would be completely incompatible with the Webpack template configuration, and would instead resemble `vite-plugin-electron`.
It should only have a very small amount of configuration, like the following.

- For Electron-Main, Preload-Scripts

```ts
export interface VitePluginConfig {
  /**
   * Shortcut of `build.lib.entry`
   */
  entry?: string | string[];
  vite?: string;
}
```

*forge.config.js*
```js
module.exports = {
  plugins: [
    {
      name: '@electron-forge/plugin-vite',
      config: [
        {
          entry: 'src/main.js',
          vite: './vite.main.config.js',
        },
        {
          entry: 'src/preload.js',
          vite: './vite.preload.config.js',
        },
      ],
    },
  ],
};
```

- For Electron-Renderer

There is no need to provide any configuration, we just need to provide the default `vite.config.js` file in the template, which is more in line with Vite users' usage habits

Reply:
We can write some default configuration, such as multiple entries, in `vite.config.js` provided by the template.
But it can always be easily modified by Vite users.

Reply:
I think the flat configuration you specified makes the most sense, especially if folks are coming from a Vite background.

I would love to have additional opinions here @electron/forgers.

Reply:
I agree with @erickzhao, if this is what Vite users would expect, it makes good sense to me to implement it this way 👍

Reply:
Final look :)
```js
{
  name: '@electron-forge/plugin-vite',
  config: {
    /**
     * Build anything such as Main-Process, Preload-Scripts, Worker-Process, etc.
     * @type {VitePluginConfig[]}
     */
    builds: [
      {
        // Vite 3.2.0 can support multiple `lib.entry`.
        entry: [
          'src/main.js',
          'src/main-worker.js',
        ],
        vite: 'vite.main.config.js',
      },
      {
        entry: 'src/preload.js',
        vite: 'vite.preload.config.js',
      },
    ],
    /**
     * Vite's CLI Options, for serve and build.
     * @type {Record<string, string>}
     */
    CLIOptions: {},
  },
}
```

Reply:
I think this works well as long as we properly document everything! @caoxiemeihao